tools/topology/topology2/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: BSD-3-Clause
+add_custom_target(topologies2 ALL)
+
+add_dependencies(topologies2 topologies1)
+
+# check alsatplg version and return if version is less than 1.2.6
+execute_process(COMMAND alsatplg --version OUTPUT_VARIABLE ALSATPLG_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+string(REGEX MATCH "[0-9].[0-9].[0-9]" VERSION_NUMBER ${ALSATPLG_VERSION})
+if(${VERSION_NUMBER} VERSION_LESS "1.2.6")
+	message(WARNING "Cannot build topology2. Minimum required version for alsatplg: 1.2.6")
+	return()
+endif()
+
+add_subdirectory(cavs)
```

Comment:
You don't need to omit the entire subdirectory, instead omit only the edge in the dependency graph and don't add this target to the default ALL target. In other words, only this line needs to be conditional:
```
add_dependencies(topologies2 topology2_cavs)
```

So everyone is aligned and runs the exact same CMake code at configuration time no matter what their ALSA version is. The difference comes later only at build time.

This is also more flexible for any adventurer who has multiple ALSA versions installed.

Reply:
but as far as topology2 goes everything in the topology2 folder should be conditionally built based on version right? its not just topology2_cavs

Reply:
Yes sorry, hasty review and copy/paste. The point is to use a scalpel to conditionally cut only the topology2 dependency in the graph, then all v2 dependency "leaves" will be configured and available but not built by default.

Reply:
so this look good?

Reply:
No, instead of `return()` here use:
```
add_dependencies(ALL topologies2)
```
\+ remove ALL above